test/e2e/e2e_test.go

```diff
@@ -0,0 +1,157 @@
+	testfiles.AddFileSource(testfiles.RootFileSource{Root: "../.."})
+}
+
+var fileSystemID = flag.String("file-system-id", "", "required. file system ID of the EFS file system to use for the test")
+
+func TestEFSCSI(t *testing.T) {
+	if *fileSystemID == "" {
+		log.Fatalf("file-system-id required. file system ID of the EFS file system to use for the test.")
+	}
+	gomega.RegisterFailHandler(ginkgo.Fail)
+
+	// Run tests through the Ginkgo runner with output to console + JUnit for Jenkins
+	var r []ginkgo.Reporter
+	if framework.TestContext.ReportDir != "" {
+		if err := os.MkdirAll(framework.TestContext.ReportDir, 0755); err != nil {
+			log.Fatalf("Failed creating report directory: %v", err)
+		} else {
+			r = append(r, reporters.NewJUnitReporter(path.Join(framework.TestContext.ReportDir, fmt.Sprintf("junit_%v%02d.xml", framework.TestContext.ReportPrefix, config.GinkgoConfig.ParallelNode))))
+		}
+	}
+	log.Printf("Starting e2e run %q on Ginkgo node %d", framework.RunID, config.GinkgoConfig.ParallelNode)
+
+	ginkgo.RunSpecsWithDefaultAndCustomReporters(t, "EFS CSI Suite", r)
+}
+
+type efsDriver struct {
+	driverInfo testsuites.DriverInfo
+}
+
+var _ testsuites.TestDriver = &efsDriver{}
+
+// TODO implement Inline (unless it's redundant) and DynamicPV
+// var _ testsuites.InlineVolumeTestDriver = &efsDriver{}
+var _ testsuites.PreprovisionedPVTestDriver = &efsDriver{}
+
+func InitEFSCSIDriver() testsuites.TestDriver {
+	return &efsDriver{
+		driverInfo: testsuites.DriverInfo{
+			Name:                 "efs.csi.aws.com",
+			SupportedFsType:      sets.NewString(""),
+			SupportedMountOption: sets.NewString("tls"),
+			Capabilities: map[testsuites.Capability]bool{
+				testsuites.CapPersistence: true,
+				testsuites.CapExec:        true,
+				testsuites.CapMultiPODs:   true,
+				testsuites.CapRWX:         true,
+			},
+		},
+	}
+}
+
+func (e *efsDriver) GetDriverInfo() *testsuites.DriverInfo {
+	return &e.driverInfo
+}
+
+func (e *efsDriver) SkipUnsupportedTest(testpatterns.TestPattern) {}
+
+func (e *efsDriver) PrepareTest(f *framework.Framework) (*testsuites.PerTestConfig, func()) {
+	ginkgo.By("Deploying EFS CSI driver")
+	cancelPodLogs := testsuites.StartPodLogs(f)
+
+	/* TODO deploy the driver before the test.
+	cleanup, err := f.CreateFromManifests(nil, "deploy/kubernetes/manifest.yaml")
+	if err != nil {
+		framework.Failf("Error deploying EFS CSI driver: %v", err)
+	}
+	*/
+
+	return &testsuites.PerTestConfig{
+			Driver:    e,
+			Prefix:    "efs",
+			Framework: f,
+		}, func() {
+			ginkgo.By("Cleaning up EFS CSI driver")
+			/* TODO deploy the driver before the test.
+			cleanup()
+			*/
+			cancelPodLogs()
+		}
+}
+
+func (e *efsDriver) CreateVolume(config *testsuites.PerTestConfig, volType testpatterns.TestVolType) testsuites.TestVolume {
+	// TODO it's assumed that the EFS volumeHandle referred to in GetPersistentVolumeSource already exists. Instead, it could be dynamically created per test here?
```

Comment:
How about creating EFS filesystem inside here?

Reply:
I decided not to as there isn't much point. It will take a long time to create one per test and there's no need for every test to use its own EFS filesystem, it's fine for them to reuse the same one

Reply:
Makes sense